### ISSUE TITLE
Add the possibility to ignore some constructors in no-new

### ISSUE THREAD

**Issue by leipert** [eslint/eslint#10484]
**What rule do you want to change?**

`no-new`

**Does this change cause the rule to produce more or fewer warnings?**

- fewer

**How will the change be implemented? (New option, new default behavior, etc.)?**

- By default all side-effect usages of `new` will be reported, like now
- You have the possibility to add constructors to an option, which will then be allowed. 

Like: 
```yml
no-new:
  - error
  - ignore:
      - Vue
```

**Please provide some example code that this change will affect:**

We often initialize Vue components with this pattern:

```js
import Component from './components/component';

document.addEventListener('DOMContentLoaded', () => {
  const element = document.getElementById('id');
  new Vue({
    el: element,
    components: {
      Component,
    },
    render(createElement) {
      return createElement('component', {
        props: { ...element.dataset },
      });
    },
   })
})
```

See our open issues here:
- https://gitlab.com/gitlab-org/gitlab-ce/issues/42783 
- https://gitlab.com/gitlab-org/gitlab-ce/issues/42306

**What does the rule currently do for this code?**

Throw an error. There are several workarounds for this, that are not nice:
1. add `eslint-disable`, this really adds up if you have a lot of components.
2. return the result of `new Vue` in the function, but this makes no sense, as no function caller is actually using it

**What will the rule do after it's changed?**

Throw no new error.

**Comment by platinumazure** [comment]
I could get behind this.

Should we support ignoring just constructor identifier names? I feel it could be a bit more flexible if we instead allowed users to specify AST selectors which we would match against the callee nodes of the NewExpression.

Examples with AST selectors:

* `[name='Vue']` (ignores `new Vue()`)
* `[object.name='Backbone']` (ignores `new Backbone.foo()` for any `foo`)

But the trade-off is that the rule configuration has a little steeper learning curve.

**Comment by leipert** [comment]
Do we already use AST selectors in other rules? If yes, we should go for it. If not, we should probably go for an easier solution. I don't know too much how those ASTs do work, could we go for something like a string match on the callee?

**Comment by leipert** [comment]
Should be really easy to do that similar to: https://github.com/eslint/eslint/blob/master/lib/rules/no-new-wrappers.js#L31

**Comment by leipert** [comment]
@platinumazure I'll assign it to myself and see if I get something up and running in a PR 👍

**Comment by platinumazure** [comment]
You're right that it would be easy to do just by looking at `node.callee.name`, but that only works for single names that are not part of another object. AST selectors might be overkill for your use case, but as a member of the ESLint team, I have to look at the bigger picture and try to anticipate other reasonable use cases before we agree to take on the maintenance burden that comes with adding a new option.

You're welcome to create a PR if you wish. However, in order for us to accept a PR, we need team consensus that the change is worth accepting. Consensus is defined in this project as 1 team member championing the proposal, 3 more team members :+1:ing the proposal, and no team member :-1:ing the proposal. It is quite possible that the team will want to tweak your intended approach before accepting the proposal, and it is also possible that the proposal won't be accepted at all. So you are welcome to create a PR, but you are doing so at the risk of wasting some of your time.

**Comment by leipert** [comment]
Thanks for the in-depth explanation of a process that I surely could have looked up somewhere. I am going to wait for those 👍 coming in before I do something.

**Comment by leipert** [comment]
@platinumazure Sorry to bring it up again, but as I mentioned, I am new here: What do I need to do to lobby more for it, is it appropriate to ping the other eslint TSC members/developers?

**Comment by platinumazure** [comment]
@eslint/eslint-team This needs a champion and one more :+1:, thanks!

**Comment by not-an-aardvark** [comment]
Is anyone willing to champion this issue? If not, it might be time to close it.

**Comment by minomikula** [comment]
We would love to have this functionality. We use this pattern:
`  var button1 = new Button({ id: 'botton1', click: handleBth1, ...});
  new Button({ id: 'botton2', click: handleBth2, ...});
  ..
  button1.disable();
`
So we have some methods on instances of Button, but not in every scenario we need to keep instance.
Currently we disable this rule for those lines, but it is not ideal.
